src/Components/popup.luau

```diff
@@ -179,31 +188,34 @@ function popup.addPopup(self, scope, props)
 		},
 	}
 	self.enabled:set(true)
-	newPopup.Parent = currentContainer
+	if currentContainer and currentContainer.Parent then
+		newPopup.Parent = currentContainer
+	end
+
 	self.currentPopup = newPopup
 end
 
 function popup.customPopup(self, scope, child)
 	if self.currentContainer == nil then
 		return
 	end
-	if self.currentPopup ~= nil then
-		self.currentPopup:Destroy()
-		self.currentPopup = nil
-	end
-	if self.currentCustomPopup ~= nil then
-		self.currentCustomPopup.Parent = nil
-		self.currentCustomPopup = nil
-	end
+
+	cleanupPopups(self)
+
 	local currentContainer = scope.peek(self.currentContainer)
 	self.currentCustomPopup = child
 	self.enabled:set(true)
-	child.Parent = currentContainer
+
+	if currentContainer and currentContainer.Parent then
```

Comment:
This check for currentContainer is not needed as the beginning of the function already does this check